src/test/java/org/elasticsearch/action/support/replication/ShardReplicationOperationTests.java

```diff
@@ -566,7 +779,23 @@ protected boolean resolveIndex() {
         }
     }
 
-    static class ActionWithConsistency extends Action {
+    /*
+    * Returns testIndexShard or initializes it if it was already created in this test run.
+    * */
+    private IndexShard getIndexShard(ShardId shardId) {
+        try {
+            if (testIndexShard == null) {
```

Comment:
can this be called concurrently? if so this needs synchronization otherwise we create it more than once?